src/attributes/attr.js

```diff
@@ -110,8 +110,13 @@ boolHook = {
 
 			// Remove boolean attributes when set to false
 			jQuery.removeAttr( elem, name );
-		} else {
+		} else if (
+			name.toLowerCase() !== "hidden" ||
+				String( value ).toLowerCase() !== "until-found"
+		) {
 			elem.setAttribute( name, name );
+		} else {
+			elem.setAttribute( name, value );
```

Comment:
```suggestion
        } else {
            elem.setAttribute( name, name.toLowerCase() === "hidden" &&
                strValue.toLowerCase() === "until-found" ? strValue : name );
```
(with `var strValue = String( value );`)